client/components/stats-date-control/stats-date-control-picker.tsx

```diff
@@ -29,8 +29,21 @@ const DateControlPicker = ( {
 		moment( dateRange.chartEnd ).format( 'YYYY-MM-DD' )
 	);
 	const infoReferenceElement = useRef( null );
+	const popoverRef = useRef( null );
 	const [ popoverOpened, togglePopoverOpened ] = useState( false );
 
+	useEffect( () => {
+		function handleClickOutside( event ) {
+			if ( popoverRef.current && ! popoverRef.current.contains( event.target ) ) {
+				togglePopoverOpened( false );
```

Comment:
As the [PageModuleToggler](https://github.com/Automattic/wp-calypso/blob/911fd241a11ece3fb6619b206a8dca1d304b799b/client/blocks/stats-navigation/page-module-toggler.tsx#L74-L78) did previously, we can pass `togglePopoverOpened` as the `onClose` prop method to the `Popover` component for closing when clicking outside by the built-in functionality.